src/modules/uuv_pos_control/uuv_pos_control.hpp

```diff
@@ -0,0 +1,161 @@
+#include <px4_platform_common/defines.h>
+#include <px4_platform_common/posix.h>
+#include <px4_platform_common/tasks.h>
+#include <px4_platform_common/module.h>
+#include <px4_platform_common/module_params.h>
+#include <px4_platform_common/px4_work_queue/ScheduledWorkItem.hpp>
+#include <uORB/Subscription.hpp>
+#include <uORB/SubscriptionCallback.hpp>
+#include <uORB/Publication.hpp>
+#include <uORB/topics/manual_control_setpoint.h>
+#include <uORB/topics/parameter_update.h>
+#include <uORB/topics/position_setpoint_triplet.h>
+#include <uORB/topics/vehicle_attitude.h>
+#include <uORB/topics/vehicle_local_position.h>
+#include <uORB/topics/vehicle_attitude_setpoint.h>
+#include <uORB/topics/vehicle_angular_velocity.h>
+#include <uORB/topics/vehicle_rates_setpoint.h>
+#include <uORB/topics/vehicle_control_mode.h>
+#include <uORB/topics/actuator_controls.h>
+#include <uORB/uORB.h>
+
+using matrix::Eulerf;
+using matrix::Quatf;
+using matrix::Matrix3f;
+using matrix::Vector3f;
+using matrix::Dcmf;
+
+using uORB::SubscriptionData;
+
+using namespace time_literals;
+
+class UUVPOSControl: public ModuleBase<UUVPOSControl>, public ModuleParams, public px4::WorkItem
+{
+public:
+	UUVPOSControl();
+	~UUVPOSControl();
+
+	/** @see ModuleBase */
+	static int task_spawn(int argc, char *argv[]);
+
+	static int custom_command(int argc, char *argv[]);
+
+	/** @see ModuleBase */
+	static int print_usage(const char *reason = nullptr);
+
+	bool init();
+
+private:
+	uORB::Publication<vehicle_attitude_setpoint_s> _att_sp_pub{ORB_ID(vehicle_attitude_setpoint)};
+
+	uORB::SubscriptionInterval _parameter_update_sub{ORB_ID(parameter_update), 1_s};
+
+	uORB::Subscription _vehicle_attitude_sub{ORB_ID(vehicle_attitude)};	/**< current vehicle attitude */
+	uORB::Subscription _manual_control_setpoint_sub{ORB_ID(manual_control_setpoint)};	/**< notification of manual control updates */
+	uORB::Subscription _vcontrol_mode_sub{ORB_ID(vehicle_control_mode)};		/**< vehicle status subscription */
+	uORB::Subscription _pos_sp_triplet_sub{ORB_ID(position_setpoint_triplet)}; /**< position setpoint */
+
+	uORB::SubscriptionCallbackWorkItem _vehicle_local_position_sub{this, ORB_ID(vehicle_local_position)};
+
+
+	//actuator_controls_s _actuators {}; /**< actuator control inputs */
+	manual_control_setpoint_s _manual_control_setpoint {}; /**< r/c channel data */
+	vehicle_attitude_s _vehicle_attitude {}; /**< vehicle attitude */
+	position_setpoint_triplet_s _pos_setpoint {}; /**< vehicle position setpoint */
+	vehicle_control_mode_s _vcontrol_mode {}; /**< vehicle control mode */
+
+	perf_counter_t	_loop_perf; /**< loop performance counter */
+
+	DEFINE_PARAMETERS(
+		// direct access to inputs
+		(ParamFloat<px4::params::UUV_DES_X>) _param_direct_pos_x, /**< desired X Position */
+		(ParamFloat<px4::params::UUV_DES_Y>) _param_direct_pos_y, /**< desired Y Position */
+		(ParamFloat<px4::params::UUV_DES_Z>) _param_direct_pos_z, /**< desired Z Position*/
```

Comment:
Passing desired position via a parameter seems like the wrong approach and unsafe. Is there a reason why this is not coming from the position setpoints?

Reply:
What you are trying to do is controlling basically the altitude of the vehicle. Therefore why not just implement it on altitude mode?

Reply:
yes you are right. This is an unsafe approach. But in the Lab often it is desired to test out different positions, or the right behavior, without connecting a companion computer.

Do you have a better solution for this? Maybe using the default parameter to prevent using these parameters directly?  Or only use the Position controller if activated?

Reply:
@Zarbokk For that we have the offboard interface. You can send your desired position setpoints through mavlink

Reply:
That's what we do with the companion computer.

But if I remember correctly, it is not possible to just send one command. The Offboard requires a constant framerate, correct?
This is not possible in an Underwater scenario, where no cable is connected. Therefore in the past we used these parameters to stabilize the robot. 

I can delete the parameter. Maybe a manual control with a game pad is a good idea?

Reply:
@Zarbokk For that kind of use case, you have the `MAV_CMD` interface that works in mission mode.

For using a manual control, it should be in `altitude` mode than manual mode where you control this IMHO but lets see what works for you

Reply:
I do not think the mission mode would be appropriate in this example.
For a "good" implementation we need to include a depth estimation module based on an external depth sensor.
Then a very good stabilization controller can be realized, which holds the depth/pitch/roll and no input to yaw.

For now, I am going to remove the parameter since this is not a good solution.
The only way to communicate with the position module, for now, is through offboard control. (position setpoints and pitch+roll = 0)

@DanielDuecker Can you have a look at this package?